### ISSUE TITLE
Mediator Diagnostics detects other AddMediator call as belonging to itself

### ISSUE THREAD

**Issue by Blackclaws** [martinothamar/Mediator#80]
This is a weird one and definitely non-standard usage. So I'm using a _second_ mediator package in my codebase right now that also has an AddMediator call on the service collection. The issue is that it has options that are configured. The Mediator Analyzer picks this up and errors out. It should probably check the namespace of the AddMediator extension call before printing an error as it might be a different Mediator framework that's being used.

**Comment by martinothamar** [comment]
Ahh, yes you are right, it needs to check the namespace but also that the assembly is the current assembly (the one from the current compilation/context). Will let you know when its fixed, thanks for the report!

**Comment by martinothamar** [comment]
Hey, this should be fixed/unblocked now in the latest 2.1 preview. So the following should work:

```csharp
services.AddMediator((Mediator.MediatorOptions opts) => opts.ServiceLifetime = ServiceLifetime.Scoped);
```
